### ISSUE TITLE
torch.mm in CPU with inconsistent results

### ISSUE THREAD

**Issue by ldmoser** [pytorch/pytorch#23538]
I am using torch.mm to transform 3D points by a given 3x3 matrix and I get different results depending whether I multiply each point independently or if I multiply them at once. The error is in the order of 1e-7 for float32 precision. Considering the inputs are exactly the same, and I'm doing it in CPU, I don't see why the output would differ. Also, if I send the tensors to GPU the problem goes away.

To Reproduce, you can use code below:

useCuda = False
import numpy as np
import torch
np.random.seed(1412)
nPoints = 2
points = torch.tensor(np.random.randn(nPoints,3),dtype=torch.float32)
R = torch.tensor(np.random.randn(3,3),dtype=torch.float32)
if useCuda:
... points = points.cuda()
... R = R.cuda()
result1 = torch.mm(R,points.t()).t()
result2 = torch.cat([torch.mm(R,points[i,:,None]).t() for i in xrange(nPoints)] )
assert( torch.max(torch.abs(result1-result2)), 1e-20 )

Note that I'm creating a random matrix, because it's coming from an internal unit test. I would imagine it's not sensitive to the specific matrix values.

## Expected behavior

I expect it to pass the assert().
Note that changing useCuda=True the assert passes.

## Environment

 - PyTorch Version:  1.0 (Cuda 9.1)
 - OS : Linux - Cent6
 - How you installed PyTorch: source
 - Python version: 2.7
 - CUDA/cuDNN version: 9.1
 - GPU models and configuration: N/A

**Comment by soumith** [comment]
1e-7 for float32 precision is totally game.
Even if your inputs are the same, the order in which the operations are done likely isn't, and floating point error accumulates

**Comment by ldmoser** [comment]
how would you explain that the GPU implementation doesn't show that problem?
